### PR TITLE
Generate passage-only items from the manifest (WBL-66)

### PR DESCRIPTION
**Pull request by rakeshqdsa**
This is for converting QTI to Learnosity.

We want to generate items with passages only, from the manifest. There will be a <resource> element that links to a separate HTML file (the passage) in the content package.
Metadata
As these are still Learnosity “items”, the LOM should be processed for tags as they are with regular XML items to convert.

Command flag
Can we have this functionality be opt-in? So, if a flag is passed to the command in the terminal, the library will also convert passages to items. Something like --passage-only-items

Expected output (JSON)
 

```
{
    "item": {
        "reference": "LEAR-VG-b107bfee-a18e-4753-ab9a-f2c18ce058a9",
        "content": "<div class=\"row\"><div class=\"col-xs-12\"><span class=\"learnosity-feature feature-50acf8bd-61fd-411a-aeae-85c3550aee22\"></span></div></div>",
        "status": "published",
        "definition": {
            "widgets": [
                {
                    "reference": "50acf8bd-61fd-411a-aeae-85c3550aee22"
                }
            ],
            "template": "dynamic"
        },
        "tags": []
    },
    "questions": [],
    "features": [
        {
            "type": "sharedpassage",
            "widget_type": "feature",
            "reference": "50acf8bd-61fd-411a-aeae-85c3550aee22",
            "data": {
                "type": "sharedpassage",
                "content": "<p>aScelerisque vestibulum</p>"
            },
            "metadata": {
                "name": "Passage",
                "template_reference": "d5d43bd6-d02a-4969-a79a-e10b344549a8"
            }
        }
    ]
}
```

### CODE REVIEW

**Comment by michaelsharman** [comment]
The converted JSON is not correct. This is what is being currently generated:

```
{
    "item": {
        "reference": "70781ce0-d994-4415-82eb-7197aa7bc2f2",
        "status": "published",
        "template": "dynamic",
        "tags": [],
        "questions": [],
        "features": [],
        "definition": {
            "scroll": {
                "enabled": true
            },
            "widgets": []
        }
    },
    "questions": [],
    "features": [
        {
            "type": "sharedpassage",
            "widget_type": "feature",
            "reference": "659c68a9-c723-4af9-be50-7ec100fc6755",
            "data": {
                "type": "sharedpassage",
                "content": "passage content here"
            },
            "item_reference": ""
        }
    ]
}
```

Instead what we need is:

```
{
    "item": {
        "reference": "70781ce0-d994-4415-82eb-7197aa7bc2f2",
        "status": "published",
        "definition": {
            "widgets": [
                {
                    "reference": "659c68a9-c723-4af9-be50-7ec100fc6755"
                }
            ],
            "template": "dynamic"
        },
        "tags": []
    },
    "questions": [],
    "features": [
        {
            "type": "sharedpassage",
            "widget_type": "feature",
            "reference": "659c68a9-c723-4af9-be50-7ec100fc6755",
            "data": {
                "type": "sharedpassage",
                "content": "passage content here"
            }
        }
    ]
}
```

Differences are:

- `definition.widgets` is empty
- `questions` and `features` are incorrectly duplicated inside `item`
- `item_reference` is incorrectly included in `features`
- `item.template` should be in `item.definition`

**Comment by rakeshqdsa** [comment]
All the mentioned issue has been fixed except second as this is not reproducible .

File: src/Services/ConvertToLearnosityService.php [hunk 1]
```diff
@@ -369,22 +382,22 @@ private function getTaxonPathEntryForItemTags(\DOMNode $resource)
...
                 $this->output->writeln("<info>" . static::INFO_OUTPUT_PREFIX . "Not an <assessmentItem>, moving on...</info>");
                 return $results;
             }
 
             $resourcePath = $currentDir . '/' . $resourceHref;
-            $results = $this->convertAssessmentItem($xmlString, $itemReference, $resourcePath, $metadata, $itemTagsArray);
-
-        } catch (\Exception $e) {
+            $results = $this->convertAssessmentItem($xmlString, $itemReference, $resourcePath, $metadata, $itemTagsArray, $resourceType);
+            } catch (\Exception $e) {
```
Comment (jack-lin-lrn) [inline src/Services/ConvertToLearnosityService.php]: Could you please move backwards in order to align the indent?

Comment (rakeshqdsa) [reply src/Services/ConvertToLearnosityService.php]: Done

**Comment by michaelsharman** [comment]
Apologies but I made a mistake in the expected JSON. We need to include a `questions` and `features` key in the main `item` body. These keys will link the reference to the question or feature from within the `definition`. For this ticket, `questions` would be an empty array and `features` would hold the sharedpassage reference.  Eg:

`{
    "item": {
        "reference": "70781ce0-d994-4415-82eb-7197aa7bc2f2",
        "status": "published",
        "definition": {
            "widgets": [
                {
                    "reference": "659c68a9-c723-4af9-be50-7ec100fc6755"
                }
            ],
            "template": "dynamic"
        },
        "questions": [],
        "features": [
            {
                "reference": "659c68a9-c723-4af9-be50-7ec100fc6755"
            }
        ],
        "tags": []
    },
    "questions": [],
    "features": [
        {
            "type": "sharedpassage",
            "widget_type": "feature",
            "reference": "659c68a9-c723-4af9-be50-7ec100fc6755",
            "data": {
                "type": "sharedpassage",
                "content": "passage content here"
            }
        }
    ]
}`

We've created an integration branch to consolidate the PRs. Could you please fork this branch and create a new PR from that?

The new branch to fork from is https://github.com/Learnosity/learnosity-qti/tree/release/v2.0.0-alpha

**Comment by ttton** [comment]
Hi @rakeshqdsa, did you get a chance to look at the comment @jack-lin-lrn left here?

> According to Michael Sharman's latest comment, we need to include questions and features in
item body for this branch which doesn't include them currently.

If this is already fixed on the integration branch, then all good.

**Comment by rakeshqdsa** [comment]
Yes this is already fixed on integration branch.

**Comment by ttton** [comment]
I haven't heard any response here, so I'm going to close this PR and make a note of this issue separately.

**Pull request closed without merging**